shared/lib/sharing/publicize/actions.js

```diff
@@ -0,0 +1,69 @@
+/**
+ * Internal dependencies
+ */
+import wpcom from 'lib/wp';
+import {
+	FETCH_PUBLICIZE_CONNECTIONS,
+	RECEIVE_PUBLICIZE_CONNECTIONS,
+	FAIL_PUBLICIZE_CONNECTIONS_REQUEST
+} from 'lib/redux-store/action-types';
+
+/**
+ * Triggers a network request to fetch Publicize connections for the specified
+ * site ID.
+ *
+ * @param  {Number}   siteId Site ID
+ * @return {Function}        Action thunk
+ */
+export function fetchConnections( siteId ) {
+	return ( dispatch ) => {
+		dispatch( {
+			type: FETCH_PUBLICIZE_CONNECTIONS,
+			siteId
+		} );
+
+		return new Promise( ( resolve ) => {
```

Comment:
I'm pretty much in favor of `redux-thunk` as a means for good async. While it's not the purest, it's pure enough, it's probably the most tested Redux-middleware-based approach out there, and it's one that all of us can generally wrap our heads around the easiest. We can always revise later.